doc/source/docker/run-as-subprocess.rst

```diff
@@ -1,53 +1,113 @@
-Run ClientApp as a Subprocess
-=============================
+Run ClientApp or ServerApp as a Subprocess
+==========================================
 
-In this mode, the ClientApp is executed as a subprocess within the SuperNode Docker
-container, rather than running in a separate container. This approach reduces the number
-of running containers, which can be beneficial for environments with limited resources.
-However, it also means that the ClientApp is no longer isolated from the SuperNode,
-which may introduce additional security concerns.
+The SuperLink and SuperNode components support two distinct process modes, allowing for
+flexible deployment and control:
 
-Prerequisites
--------------
+1. Subprocess Mode: In this configuration (default), the SuperLink and SuperNode take
+   responsibility for launching the Server and ClientApp processes internally. This
+   differs from process mode ``isolation`` by using separate containers, as demonstrated
+   in the :doc:`tutorial-quickstart-docker` guide.
 
-1. Before running the ClientApp as a subprocess, ensure that the FAB dependencies have
-   been installed in the SuperNode images. This can be done by extending the SuperNode
-   image:
+   This approach reduces the number of running containers, which can be beneficial for
+   environments with limited resources. However, it also means that the applications are
+   not isolated from their parent containers, which may introduce additional security
+   concerns.
 
-   .. code-block:: dockerfile
-       :caption: Dockerfile.supernode
-       :linenos:
-       :substitutions:
+2. Isolation Mode: In this mode, the Server and ClientApp run in completely separate
+   processes. Unlike the alternative Subprocess mode, the SuperLink or SuperNode does
+   not attempt to create or manage these processes. Instead, they must be started
+   externally.
 
-       FROM flwr/supernode:|stable_flwr_version|
+These modes can be mixed for added flexibility. For instance, you can run the SuperLink
+in ``subprocess`` mode while keeping the SuperNode in ``isolation`` mode, or vice versa.
+It is also possible to configure both to run entirely in ``isolation`` mode, depending
+on your specific requirements.
```

Comment:
I think this sentence is now unnecessary with the changes above.

Reply:
removed